### PR TITLE
Nano Fabricator Repairs

### PR DESCRIPTION
**Pull request by bombasticSlacks**
#### Summary
Content "High end items can be repaired in a Nano Fabricator"

#### Purpose of change
One of the ways to make the Phase Immersion Suit repairable. Figured I would generalize to other tools.

#### Describe the solution
Added a new item flag *NANOFAB_REPAIR*. This flag describes the item as having a holographic patch that would have, in game, been created by the nanofabricator on printing the item originally and inscribes the details and procedures to repair it.

This flag is added to items when you print them with a nanofabricator *or* can be added to items base info so they are repairable in a nanofab by default.

If an item with the flag is put in the Nanofabricator bed it identifies it and instead of trying to print something else asks if you would like to repair it (or says it doesn't need repairs). 

Repairs cost 5% of printing costs per level of damage the item has taken.

#### Describe alternatives you've considered
I'll be PRing another solution to work with this soon.

#### Testing
Still drafted just wanted to add screen shots and stuff 

#### Additional context
image

image

image

image

### CODE REVIEW

**Comment by schoolcf** [comment]
if it does not speak, does an illiterate character just intrinsically know how to use a nano fabricator
should probably either make it make noise/talk or check reading

**Comment by bombasticSlacks** [comment]
> if it does not speak, does an illiterate character just intrinsically know how to use a nano fabricator should probably either make it make noise/talk or check reading

I assumed the above text prompts would display on the monitor as well as be accompanied by text to speech voice.

**Comment by schoolcf** [comment]
> > if it does not speak, does an illiterate character just intrinsically know how to use a nano fabricator should probably either make it make noise/talk or check reading
> 
> I assumed the above text prompts would display on the monitor as well as be accompanied by text to speech voice.

but it doesn't make noise

File: src/iexamine.cpp [hunk 1]
```diff
@@ -348,43 +349,98 @@ void iexamine::change_appearance( Character &you, const tripoint & )
...
+    requirement_data reqs;
 
-    if( !nanofab_template ) {
-        return;
+    // If there is something on the table that can be repaired suggest to repair that instead of printing something new
+    if( !on_table.empty() ) {
+        new_item = item( on_table.front()->typeId(), calendar::turn );
+        int damage = on_table.front()->damage_level();
+        if( damage <= 0 ) {
+            popup( _( "FABRICATOR COMPLIANT ITEM %s DETECTED, BUT NOT DAMAGED REMOVE WORKING ITEM FROM FABRICATOR" ),
```
Comment (Faalagorn) [inline src/iexamine.cpp:379]: Are the lacking full stops (dots) here intentional?

File: src/iexamine.cpp [hunk 2]
```diff
@@ -348,43 +349,98 @@ void iexamine::change_appearance( Character &you, const tripoint & )
...
+                   new_item.display_name() );
+            sounds::sound( spawn_point, 10, sounds::sound_t::speech,
+                           _( "REMOVE WORKING ITEM FROM FABRICATOR." ), true );
+            return;
+        }
+
+        if( !on_table.front()->empty() ) {
+            popup( _( "ITEM %s DETECTED, CONTENTS OF ITEM WILL BE DESTROYED BY FABRICATOR PLEASE EMPTY ITEM AND RETURN" ),
```
Comment (Faalagorn) [inline src/iexamine.cpp:387]: Are the lacking full stops (dots) here intentional? (same as above)

**Review by Faalagorn — COMMENTED** [review]
Minor issues I noticed on Transifex.

File: src/iexamine.cpp (hunk 1 above)
Comment (bombasticSlacks) [reply src/iexamine.cpp:379]: Iirc other all caps style announcements don't have dots but I might be wrong.

**Pull request merged**